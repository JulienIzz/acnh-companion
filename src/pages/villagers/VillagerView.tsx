import React from 'react';
import {View, Image} from 'react-native';
import {SimpleTypography} from '../../theme/typography/Typography';
import {StyleSheet} from 'react-native';
import {Villager} from './VillagerType';

interface Props {
  villager: Villager;
}

export const VillagerView = ({villager}: Props) => {
  return (
    <View style={styles.animalCard}>
      <View style={styles.animalID}>
        <SimpleTypography text={'Nom : ' + villager.name['name-EUfr']} />
        <SimpleTypography text={'Personnalité : ' + villager.personality} />
        <SimpleTypography
          text={'Anniversaire : ' + villager['birthday-string']}
        />
        <SimpleTypography text={'Sexe : ' + villager.gender} />
        <SimpleTypography text={'Espèce : ' + villager.species} />
      </View>
      <Image
        style={styles.animalImage}
        source={{
          uri: villager.icon_uri,
        }}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  animalCard: {
    backgroundColor: '#FAEDCD',
    borderColor: '#D4A373',
    borderRadius: 10,
    borderWidth: 3,
    marginVertical: 5,
    marginHorizontal: 10,
    padding: 8,
    flexDirection: 'row',
    alignItems: 'center',
  },
  animalID: {
    flex: 1,
    justifyContent: 'space-evenly',
  },
  animalImage: {
    width: 80,
    height: 80,
    margin: 5,
  },
});
