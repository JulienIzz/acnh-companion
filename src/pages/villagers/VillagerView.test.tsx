import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {VillagerView} from './VillagerView';
import {mockVillagerArray} from '../../tests/mockVillagerArray';

test('Villager view card with mock information', async () => {
  render(<VillagerView villager={mockVillagerArray[0]} />);

  console.log(mockVillagerArray[0]);

  const nameOutput = screen.getByText('Nom : Villager1');
  const personalityOutput = screen.getByText('Personnalité : Smug');
  const birthdayOutput = screen.getByText('Anniversaire : January 1st');
  const sexOutput = screen.getByText('Sexe : Male');
  const specieOutput = screen.getByText('Espèce : Koala');

  expect(nameOutput).toBeTruthy();
  expect(personalityOutput).toBeTruthy();
  expect(birthdayOutput).toBeTruthy();
  expect(sexOutput).toBeTruthy();
  expect(specieOutput).toBeTruthy();

  expect(screen.toJSON()).toMatchSnapshot();
});
