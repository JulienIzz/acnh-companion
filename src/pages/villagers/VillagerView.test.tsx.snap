// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager view card with mock information 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FAEDCD",
      "borderColor": "#D4A373",
      "borderRadius": 10,
      "borderWidth": 3,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "marginVertical": 5,
      "padding": 8,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      Nom : Villager1
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      Personnalité : Smug
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      Anniversaire : January 1st
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      Sexe : Male
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 14,
        }
      }
    >
      Espèce : Koala
    </Text>
  </View>
  <Image
    source={
      {
        "uri": "https://acnhapi.com/v1/icons/villagers/1",
      }
    }
    style={
      {
        "height": 80,
        "margin": 5,
        "width": 80,
      }
    }
  />
</View>
`;
